python/ray/air/_internal/util.py

```diff
@@ -45,12 +52,15 @@ def skip_exceptions(exc: Optional[Exception]) -> Exception:
         # If this is a StartTraceback, skip
         return skip_exceptions(exc.__cause__)
 
-    # Else, make sure nested exceptions are properly skipped
+    # Perform a shallow copy to prevent recursive __cause__/__context__.
+    new_exc = copy.copy(exc).with_traceback(exc.__traceback__)
```

Comment:
Note: `with_traceback` is needed so that the traceback shows the original line that errored, rather than this line where the copy is happening.

Reply:
Does making a shallow copy remove the `__context__` so that the `new_exc` has no context?

Reply:
Yes, but more importantly the `__context__` gets set (to the `StartTraceback`) after the exception gets raised, so we want to make sure there is no nested `__cause__` or `__context__` that points back to this exception.